src/util.js

```diff
@@ -276,6 +276,45 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+var _hookIdCounter = 0;
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
```

Comment:
I assume you are saying this in terms of performance. If so, I will write a short benchmark, comparing "function wrapping" method and the current one. If the current method is significantly worse, than I can use "function wrapping" and rebuild the hook handler function on each "unhook".